app/controllers/DepartureTestMessagesController.scala

```diff
@@ -50,15 +52,27 @@ class DepartureTestMessagesController @Inject()(cc: ControllerComponents,
               getResponse.status match {
                 case status if is2xx(status) =>
                   departureConnector
-                    .post(request.testMessage.messageType, request.generatedMessage.toString(), departureId)
-                    .map {
-                      postResponse =>
-                        postResponse.status match {
-                          case status if is2xx(status) =>
-                            Created
-                          case _ =>
-                            handleNon2xx(postResponse)
-                        }
+                    .getMessages(departureId)
+                    .flatMap {
+                      case Right(departureWithMessages: DepartureWithMessages) =>
+                        val message = populateMesRecMES6(request.generatedMessage, "IE015", departureWithMessages.messages)
```

Comment:
A similar extracting case as the one above

Reply:
I'll add MessageType declarations to address this.